docs/cloud/guides/security/data_masking.md

```diff
@@ -0,0 +1,177 @@
+┌─user_id─┬─name─────────┬─email──────────────┬─phone────────┬─total_amount─┬─order_date─┬─shipping_address──────────┐
+│    1001 │ John ****    │ jo****@gmail.com   │ 555-***-4567 │       299.99 │ 2024-01-15 │ *** New York, NY 10001    │
+│    1002 │ Sarah ****   │ sa****@outlook.com │ 555-***-6543 │        149.5 │ 2024-01-16 │ *** Los Angeles, CA 90210 │
+│    1003 │ Michael **** │ mb****@company.com │ 555-***-7890 │          599 │ 2024-01-17 │ *** Chicago, IL 60601     │
+│    1004 │ Emily ****   │ em****@yahoo.com   │ 555-***-0987 │        89.99 │ 2024-01-18 │ *** Houston, TX 77001     │
+│    1005 │ David ****   │ dw****@email.net   │ 555-***-3210 │       449.75 │ 2024-01-19 │ *** Phoenix, AZ 85001     │
+└─────────┴──────────────┴────────────────────┴──────────────┴──────────────┴────────────┴───────────────────────────┘
+```
+
+Notice that the data returned from the view is partially masked, obfuscating the sensitive information.
+You can also create multiple views, with differing levels of obfuscation depending on the level of privileged access to information the viewer has.
+
+To ensure that users are only able to access the view returning the masked data, and not the table with the original unmasked data, you should use [Role Based Access Control](/cloud/security/cloud-access-management/overview) to ensure that specific roles only have grants to select from the view.
+
+First create the role:
+
+```sql
+CREATE ROLE masked_orders_viewer;
+```
+
+Next grant `SELECT` privileges on the view to the role:
+
+```sql
+GRANT SELECT ON masked_orders TO masked_orders_viewer;
+```
+
+Because ClickHouse roles are additive, you must ensure that users who should only see the masked view do not have any `SELECT` privilege on the base table via any role.
+
+As such, you should explicitly revoke base-table access to be safe:
+
+```sql
+REVOKE SELECT ON orders FROM masked_orders_viewer;
+```
+
+Finally, assign the role to the appropriate users:
+
+```sql
+GRANT masked_orders_viewer TO your_user;
+```
+
+This ensures that users with the `masked_orders_viewer` role are only able to see
+the masked data from the view and not the original unmasked data from the table.
+
+## Use query masking rules for log data {#use-query-masking-rules}
+
+For users of ClickHouse OSS wishing to mask log data specifically, you can make use of [query masking rules](/operations/server-configuration-parameters/settings#query_masking_rules) (log masking) to mask data.
+
+To do so, you can define regular expression-based masking rules in the server configuration.
+These rules are applied to queries and all log messages before they are stored in server logs or system tables (such as `system.query_log`, `system.text_log`, and `system.processes`).
+
+This helps prevent sensitive data from leaking into **logs** only.
+Note that it does not mask data in query results.
+
+For example, to mask a social security number, you could add the following rule to your server configuration:
```

Comment:
Maybe tell them where this config needs to go. I think usually under `config.d` would get picked up?

Reply:
or maybe just link to this page: 

https://clickhouse.com/docs/operations/configuration-files